test/iterable_basics_test.dart

```diff
@@ -371,6 +388,23 @@ void main() {
       expect(emptyInts.minBy((a) => a.toString().length), isNull);
       expect(emptyStrings.minBy((a) => a.length), isNull);
     });
+
+    test('does not calculate any sort key more than once', () {
+      final values = [3, 222, 11, 15, 18];
```

Comment:
It might make sense to have a duplicate number in these tests.

```dart
[1, 1, 2, 3].minBy((e) => e);
```

Should only return have 3 call counts, right? the second `1` will just refer to the cache?

Reply:
Good call. Updated the tests to cover having duplicates.